### PR TITLE
Fixes #34.

### LINKED ISSUE #34: NPE thrown when File#listFiles throws null
[File#listFiles](https://github.com/sbtourist/Journal.IO/blob/master/src/main/java/journal/io/api/Journal.java#L173) throwing null is not consistently handled and might throw a NPE. Fixing that in current codebase makes next #write call throw a FileNotFoundException (which is a good as before it failed silently).

Now I am wondering if an Exception should not be thrown before to ensure consistency (during open?).
Does it make sense to continue when File#listFiles returns null?

### PR DESCRIPTION
**Pull request by jeluard**
Make sure open throws an Exception when File#listFiles returns null.

### CODE REVIEW

File: src/main/java/journal/io/api/Journal.java [hunk 1]
```diff
@@ -170,6 +170,10 @@ public boolean accept(File dir, String n) {
                 return dir.equals(directory) && n.startsWith(filePrefix) && n.endsWith(fileSuffix);
             }
         });
+        if (files == null) {
+            throw new IllegalStateException("Failed to access content of "+directory);
```
Comment (sbtourist) [inline src/main/java/journal/io/api/Journal.java]: What about throwing an IOException?

Comment (jeluard) [inline src/main/java/journal/io/api/Journal.java]: Updated the pull request.

**Comment by sbtourist** [comment]
Great, just one more thing before I merge: could you amend all your commits into a single one, with a meaningful message like "Make sure open throws an Exception when File#listFiles returns null.", and then force push?
Feel free to get back if you need guidance.

That will help keeping the commit history easier to understand and navigate :)

**Comment by jeluard** [comment]
Sure! Some guidance will definitively be helpful, git and I are not exactly friends :)
What would be the way to fix the current situation? Also how should I do that right the next time given I might want to change things after the first pull request?
I am fine restarting from scratch.

Thanks for the help!

**Comment by sbtourist** [comment]
Sure :)

First, if you want to commit something, and then you notice you forget to commit something else, rather than committing again, you can "overwrite" the previous commit by using git amend, like: git commit --amend -m "commit message"
This will give you a single commit out of your last previous one and what you have right now in your working directory. Then, if you already pushed your previous commit, you have to force push (--force) the amended one, but please do that ONLY if you're pushing to your own repo, otherwise collaborators will hate you forever :)

By the way, this will not help you probably in this case, as I see you have lots of commits (sorry, didn't really notice before) and git amend only works with the last previous one: so I'd suggest to start from scratch and commit/push with a meaningful message.

In the future, just always use good commit messages, and if you want to fix the last previous commit, use git amend with care.

Hope that helps :)

**Pull request closed without merging**